portal/src/util/misc.ts

```diff
@@ -0,0 +1,17 @@
+export function isArrayEqualInOrder(arr1: unknown[], arr2: unknown[]): boolean {
```

Comment:
Better to add a type argument T to this function.

```typescript
function isArrayEqualInOrder<T>(arr1: T[], arr2: T[]): boolean
```

So we will not compare apples to oranges.